scripts/abins/kpointsdata.py

```diff
@@ -57,6 +64,14 @@ def __init__(self, *, frequencies: np.ndarray, atomic_displacements: np.ndarray,
         if not (weights.dtype.num == FLOAT_ID
                 and np.allclose(weights, weights[weights >= 0])):
             raise ValueError("Invalid value of weights.")
+
+        if not isclose(np.sum(weights), 1.0):
+            if logger is None:
+                logger = mantid_logger
```

Comment:
Would it be better to do the DI unconditionally, rather than when the sum of weights is not approx 1.0?
Long-term I can see confusion / debugging sessions arising when developers have to have specific weights for their logging to work in other places

Reply:
Good point! And it's generally more conventional to put this sort of "argument handling" close to the top of a function/method block, so that is where one would go looking for it.